src/components/sidenav/sidenav.js

```diff
@@ -257,9 +266,39 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
     scope.$watch('isOpen', updateIsOpen);
 
 
+
     // Publish special accessor for the Controller instance
     sidenavCtrl.$toggleOpen = toggleOpen;
 
+    function revalidateVisibility() {
+      var lastValue = scope.isOpen;
+      if (attr.mdIsLockedOpen) return;
+
+      if (isHidden(true, true) && scope.isOpen) {
+        scope.isOpen = false;
+      } else if (!isHidden(true, true) && !scope.isOpen) {
+        scope.isOpen = true;
+      }
```

Comment:
Sorry for the late answer. Yes `updateIsOpen` (the actual method which triggers open and close animations) is called by watching `isOpen`.